packer/builder/azure/arm/config.go

```diff
@@ -308,24 +309,51 @@ func provideDefaultValues(c *Config) {
 	}
 }
 
+// Check if device login is being asked for, and is allowed.
+//
+// Device login is enabled if the user only defines SubscriptionID and not
+// ClientID, ClientSecret, and TenantID.
+//
+// Device login is not enabled for Windows because the WinRM certificate is
+// readable by the ObjectID of the App.  There may be another way to handle
+// this case, but I am not currently aware of it - send feedback.
+func isUseDeviceLogin(c *Config) bool {
+	if c.UseDeviceLogin {
```

Comment:
While grasping Mjölnir?  I'll remove the ability to the set the variable.

I see your point, and I agree with your intention.  I've now bloated this already bloated method because there's no proper way to express "private". :frowning: